src/sagemaker_server.cc

```diff
@@ -407,49 +407,27 @@ SagemakerAPIServer::SagemakeInferRequestClass::InferResponseComplete(
   }
 #endif  // TRITON_ENABLE_TRACING
 
-  if (err == nullptr) {
-    evthr_defer(infer_request->thread_, OKReplyCallback, infer_request);
-  } else {
+  if (err != nullptr) {
     EVBufferAddErrorJson(infer_request->req_->buffer_out, err);
+    // [FIXME] In http_server.cc, error handling is enhanced to reporting
+    // different error code according to the Triton error code, holding
+    // the change from SageMaker endpoint as it may not fit their SLA.
```

Comment:
do we need similar fix for vertex endpoint as well?

Reply:
I will leave the same comment in vertex endpoint. Will also need to check with the right group of people for the same before applying the change

Reply:
Actually it is harder for vertex AI.. The endpoint just call into the http_server base class's functions and it gets the same behavior already.. I would just let it be what it is and fix later if specific error code must be returned in vertex AI deployment.